pkg/monitortests/machines/watchmachines/machines.go

```diff
@@ -83,15 +83,24 @@ func startMachineMonitoring(ctx context.Context, m monitorapi.RecorderWriter, cl
 		// this is last so machine deleted shows up last when queried
 		func(machine, oldMachine *machine.Machine) []monitorapi.Interval {
 			var intervals []monitorapi.Interval
-			if machine != nil {
-				return intervals
+
+			nodeName := "<unknown>"
+			oldHasNodeRef := oldMachine != nil && oldMachine.Status.NodeRef != nil
+			newHasNodeRef := machine.Status.NodeRef != nil
+			if oldHasNodeRef {
+				nodeName = oldMachine.Status.NodeRef.Name
+			}
+
+			if newHasNodeRef {
+				nodeName = machine.Status.NodeRef.Name
```

Comment:
So getting a correct node name into the interval annotation will allow your test logic to correctly correlate this deleting machine with a node not ready, and ignore it going forward? i.e. this will cause LESS failures without having to make the test flake.

Suggest explaining some of that in the commit message + PR when you finish this up and have added the unit tests.

Reply:
I don't have a lot of real estate to work with for commit message. I added this to the description.